environment.sh

```diff
@@ -90,24 +91,35 @@ else
       # Empty string is no return on grep
       conda install jq -y
     fi
+    echo "Checking framework packages are up to date..."
     # Check if there has been an update to vivarium packages since last modification to requirements file
     # or more reccent than environment creation
-    grep @ $install_file 
-    # TODO: can we make this grep output a variable?
-    while read -r line ; do
-      # Parse each line of grep output
-      repo_info=(${line//@/ })
-      repo=${repo_info[0]}
-      repo_branch=${repo_info[2]}
-      last_commit_time=$(curl -H "Accept: application/vnd.github.v3+json" https://api.github.com/repos/ihmeuw/$repo/commits?sha=$repo_branch | jq '.["0"].commit.committer.date')
+    # Note: The lines we will return via grep will look like 'vivarium>=#.#.#' or will be of the format 
+    # 'vivarium @ git+https://github.com/ihmeuw/vivarium@SOME_BRANCH'
+    # echo $(grep -E 'vivarium|gbd|risk_distribution|layered_config' $install_file)
+    framework_packages=$(grep -E 'vivarium|gbd|risk_distribution|layered_config' $install_file)
+    echo "Grep output: $framework_packages"
+    num_packages=$(grep -E 'vivarium|gbd|risk_distribution|layered_config' -c $install_file)
+    echo "Number of framework packages: $num_packages"
+    for ((i = 1; i <= $num_packages; i++)); do
```

Comment:
In the future if at all possible I would highly advise anyone to use for instead of while loops because you don't have to deal with subprocesses and it is a lot easier to debug.